crates/miden-lib/asm/miden/auth/multisig.masm

```diff
@@ -0,0 +1,284 @@
+use.miden::account
+use.miden::auth
+use.miden::tx
+use.std::crypto::dsa::rpo_falcon512
+use.std::crypto::hashes::rpo
+use.std::word
+
+# CONSTANTS
+# =================================================================================================
+
+# Auth Request Constants
+
+# The event to request an authentication signature.
+const.AUTH_REQUEST=131087
+
+# The event emitted when a signature is not found for a required signer.
+const.UNAUTHORIZED_EVENT=131102
+
+# Storage Layout Constants
+#
+# ┌─────────────────────────────┬──────────┬──────────────┐
+# │    THRESHOLD & APPROVERS    │ PUB KEYS │ EXECUTED TXS │
+# │           (slot)            │   (map)  │    (map)     │
+# ├─────────────────────────────┼──────────┼──────────────┤
+# │              0              │    1     │      2       │
+# └─────────────────────────────┴──────────┴──────────────┘
+
+# The slot in this component's storage layout where both the signature threshold
+# and number of approvers are stored as [threshold, num_approvers, 0, 0].
+# The threshold is guaranteed to be less than or equal to num_approvers.
+const.THRESHOLD_CONFIG_SLOT=0
+
+# The slot in this component's storage layout where the public keys map is stored.
+# Map entries: [key_index, 0, 0, 0] => owner_public_key
+const.PUBLIC_KEYS_MAP_SLOT=1
+
+# The slot in this component's storage layout where executed transactions are stored.
+# Map entries: transaction_message => [is_executed, 0, 0, 0]
+const.EXECUTED_TXS_SLOT=2
+
+# Memory Constants
+
+# Local Memory Addresses
+const.NUM_OF_APPROVERS=0
+const.PUB_KEY_MAP_IDX=4
+const.CURRENT_PK=8
+const.SUCCESSFUL_VERIFICATIONS=12
```

Comment:
nit: maybe we should add suffixes to these constant to make it clear that these are locations of variables rather than variables themselves. For example: `NUM_OF_APPROVERS` -> `NUM_OF_APPROVERS_LOC`.